chromatic_fitting/models/eclipse.py

```diff
@@ -0,0 +1,505 @@
+                        starry.Map(
+                            ydeg=0,
+                            udeg=2,
+                            amp=param_i[f"{name}stellar_amplitude"],
+                            inc=90.0,
+                            obl=0.0,
+                        ),
+                        m=param_i[f"{name}stellar_mass"],
+                        r=param_i[f"{name}stellar_radius"],
+                        prot=param_i[f"{name}stellar_prot"],
+                    )
+                    star.map[1] = param_i[f"{name}limbdark1"]
+                    star.map[2] = param_i[f"{name}limbdark2"]
+
+                    planet = starry.kepler.Secondary(
+                        starry.Map(
+                            ydeg=0,
+                            udeg=0,
+                            amp=10 ** param_i[f"{name}planet_log_amplitude"],
+                            inc=90.0,
+                            obl=0.0,
+                        ),
+                        # the surface map
+                        inc=param_i[f"{name}inclination"],
+                        m=param_i[f"{name}planet_mass"],  # mass in Jupiter masses
+                        r=param_i[f"{name}planet_radius"],  # radius in Jupiter radii
+                        porb=param_i[f"{name}period"],  # orbital period in days
+                        prot=param_i[f"{name}period"],  # orbital period in days
+                        ecc=param_i[f"{name}eccentricity"],  # eccentricity
+                        w=param_i[f"{name}omega"],  # longitude of pericenter in degrees
+                        t0=param_i[f"{name}t0"],  # time of transit in days
+                        length_unit=u.R_jup,
+                        mass_unit=u.M_jup,
+                    )
+                    planet.theta0 = 180.0
+
+                    system = starry.System(star, planet)
+                    flux_model = system.flux(data.time.to_value("day"))
+                    y_model.append(flux_model)
+
+                    self.initial_flux_model_guess.append(eval_in_model(flux_model))
+
+                # (if we've chosen to) add a Deterministic parameter to the model for easy extraction/plotting
+                # later:
+                if self.store_models:
+                    Deterministic(f"{name}model", pm.math.stack(y_model, axis=0))
+
+                # add the model to the overall lightcurve:
+                if f"wavelength_{j}" not in self.every_light_curve.keys():
+                    self.every_light_curve[f"wavelength_{j}"] = pm.math.stack(
+                        y_model, axis=0
+                    )
+                else:
+                    self.every_light_curve[f"wavelength_{j}"] += pm.math.stack(
+                        y_model, axis=0
+                    )
+
+                self.model_chromatic_eclipse_flux = [
+                    self.every_light_curve[k] for k in tqdm(self.every_light_curve)
+                ]
+
+    def sample(
```

Comment:
Added this function to choose the sampling method that will work with starry (for now pymc3_ext)